src/compiler/test-file/api-based.js

```diff
@@ -66,6 +70,9 @@ export default class APIBasedTestFileCompilerBase extends TestFileCompilerBase {
 
     async _execAsModule (code, filename) {
         if (this.esm) {
+            if (semver.gte(process.version, '20.0.0'))
```

Comment:
As I see in the documentation, `register` should also work for the 18 nodejs version. Please check it